python_src/src/lamp_py/tableau/hyper.py

```diff
@@ -0,0 +1,348 @@
+        """
+
+    @abstractmethod
+    def create_parquet(self) -> None:
+        """
+        Business logic to create new Job parquet file
+        """
+
+    @abstractmethod
+    def update_parquet(self) -> bool:
+        """
+        Business logic to update existing Job parquet file
+
+        :return bool (parquet was updated)
+        """
+
+    def convert_parquet_dtype(self, dtype: pyarrow.DataType) -> SqlType:
+        """
+        SqlType.smallint(): 2 byte signed i16
+        SqlType.int(): 4 byte signed i32
+        SqlType.bigint(): 8 byte signed i64
+        """
+        dtype = str(dtype)
+        dtype_map = {
+            "int8": SqlType.small_int(),
+            "int16": SqlType.small_int(),
+            "int32": SqlType.int(),
+            "int64": SqlType.big_int(),
+            "bool": SqlType.bool(),
+            "float16": SqlType.double(),
+            "float32": SqlType.double(),
+            "float64": SqlType.double(),
+        }
+
+        map_check = dtype_map.get(dtype)
+        if map_check is not None:
+            return map_check
+
+        if dtype.startswith("date32"):
+            return SqlType.date()
+
+        if dtype.startswith("timestamp"):
+            return SqlType.timestamp()
+
+        return SqlType.text()
+
+    def upload_parquet(self) -> None:
+        """
+        Upload local parquet file to remote path
+        """
+        process_logger = ProcessLogger(
+            "hyper_job_upload_parquet",
+            source=self.local_parquet_path,
+            destination=self.remote_parquet_path,
+        )
+        process_logger.log_start()
+        fs.copy_files(
+            source=self.local_parquet_path,
+            destination=self.remote_parquet_path,
+            source_filesystem=fs.LocalFileSystem(),
+            destination_filesystem=self.remote_fs,
+        )
+        process_logger.log_complete()
+
+    def download_parquet(self) -> None:
+        """
+        Download remote parquet file to local path
+        """
+        process_logger = ProcessLogger(
+            "hyper_job_download_parquet",
+            source=self.remote_parquet_path,
+            destination=self.local_parquet_path,
+        )
+        process_logger.log_start()
+
+        if os.path.exists(self.local_parquet_path):
+            os.remove(self.local_parquet_path)
+
+        fs.copy_files(
+            source=self.remote_parquet_path,
+            destination=self.local_parquet_path,
+            source_filesystem=self.remote_fs,
+            destination_filesystem=fs.LocalFileSystem(),
+        )
+
+        process_logger.log_complete()
+
+    def max_stats_of_parquet(self) -> Dict[str, str]:
+        """
+        Create dictionary of maximum value for each column of locally saved
+        parquet file
+        """
+        # get row_groups from parquet metadata (list of dicts)
+        row_groups = pq.read_metadata(self.local_parquet_path).to_dict()[
+            "row_groups"
+        ]
```

Comment:
Pulling max column values from parquet metadata should be a much more efficient operation compared to loading a whole parquet file into memory. 

`row_groups` can be pulled out of `read_metadata`, this is a list of list of dictionary objects. 

this function produces a dictionary object where the key is the column name and the value is the `max` value from the `statistics` object